src/python/gudhi/sklearn/rips_persistence.py

```diff
@@ -0,0 +1,126 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Vincent Rouvreau
+#
+# Copyright (C) 2022 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+from .. import RipsComplex
+from sklearn.base import BaseEstimator, TransformerMixin
+
+import numpy as np
+# joblib is required by scikit-learn
+from joblib import Parallel, delayed
+
+# Mermaid sequence diagram - https://mermaid-js.github.io/mermaid-live-editor/
+# sequenceDiagram
+#     USER->>RipsPersistence: fit_transform(X)
+#     Note right of RipsPersistence: homology_dimensions=[i,j]
+#     RipsPersistence->>thread1: _tranform(X[0])
+#     RipsPersistence->>thread2: _tranform(X[1])
+#     Note right of RipsPersistence: ...
+#     thread1->>RipsPersistence: [array( Hi(X[0]) ), array( Hj(X[0]) )]
+#     thread2->>RipsPersistence: [array( Hi(X[1]) ), array( Hj(X[1]) )]
+#     Note right of RipsPersistence: ...
+#     RipsPersistence->>USER: [[array( Hi(X[0]) ), array( Hj(X[0]) )],<br/> [array( Hi(X[1]) ), array( Hj(X[1]) )],<br/> ...]
+
+
+class RipsPersistence(BaseEstimator, TransformerMixin):
+    """
+    This is a class for computing the persistence diagrams from a Vietoris-Rips complex.
+    """
+
+    def __init__(
+        self,
+        homology_dimensions,
```

Comment:
I'd rather avoid the confusing `persistence_dim_max`, `homology_dimensions` should be enough to determine if we need internally to pass True or False.
By "magic", I mean something that isn't the usual list of integers. It could be `homology_dimensions=-1`, or `None`, `'all'`, `'all_but_max'`, etc.

Reply:
Ok, I tried to summarize your proposal on [af2399d](https://github.com/GUDHI/gudhi-devel/pull/691/commits/af2399db4ce1a0927ae7776dc7db5c9daa75547f) after our discussion.